pkg/version/migrate.go

```diff
@@ -48,77 +55,162 @@ func PopulateMissingDownstreamVersions() error {
 	return nil
 }
 
-func populateMissingDownstreamVersionsForApp(tx *sql.Tx, appID string) error {
-	var maxAppVersionSequenceSQL sql.NullInt64
-	row := tx.QueryRow(`select max(sequence) from app_version where app_id = $1`, appID)
-	if err := row.Scan(&maxAppVersionSequenceSQL); err != nil {
-		return errors.Wrap(err, "failed to find current appversion max sequence in row")
+type appDownstreamVersion struct {
+	Sequence       int64
+	ParentSequence int64
+	CreatedAt      time.Time
+	VersionLabel   string
+	Status         sql.NullString
+}
+
+type appVersion struct {
+	Sequence     int64
+	UpdateCursor sql.NullString
+	CreatedAt    time.Time
+	VersionLabel string
+}
+
+func populateMissingDownstreamVersionsForApp(tx *sql.Tx, appID string, downstream *types.Downstream) error {
```

Comment:
I would _love_ a pseudocode comment here describing what will be removed, what will be added, and why